### PR TITLE
Cat animation & state machine

### PR DESCRIPTION
**Pull request by Trylat**
## Description
This change refactor most player logic to be handdle by a StateMachine. This also split animation and player logic so they can be handdle separatly. In this commit, player have 3 states (Ground, Air and Landded) that control basics commands and some animation transitions. This is the V1 and new states will need to be added in the future so it fit more what we want for this game.

### CODE REVIEW

File: Cat&Moon/player/pla65FF.tmp
Comment (drick0230) [inline Cat&Moon/player/pla65FF.tmp:1]: Is it suppose to be commited?

File: Cat&Moon/player/Script/WalkState.gd [hunk 1]
```diff
@@ -0,0 +1,48 @@
+extends State
+
+@export var air_state : State
+@export var iddle_state : State
```
Comment (drick0230) [inline Cat&Moon/player/Script/WalkState.gd:4]: typo: idle_state

File: Cat&Moon/player/StateDebugLabel.gd [hunk 2]
```diff
@@ -0,0 +1,13 @@
+extends Label
+
+@export state_machine : Sta
```
Comment (drick0230) [inline Cat&Moon/player/StateDebugLabel.gd]: Is is suppose to be State?

File: Cat&Moon/player/pla65FF.tmp
Comment (Trylat) [reply Cat&Moon/player/pla65FF.tmp:1]: Nope lol

**Review by drick0230 — APPROVED** [review]

**Pull request merged**